stream-chat-android-offline/src/test/java/io/getstream/chat/android/livedata/mockwebserver/SendMessageUseCaseTest.kt

```diff
@@ -0,0 +1,50 @@
+package io.getstream.chat.android.livedata.mockwebserver
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.client.utils.SyncStatus
+import io.getstream.chat.android.livedata.utils.isRequestBodyEqualTo
+import kotlinx.coroutines.runBlocking
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class SendMessageUseCaseTest : BaseMockWebServerIntegrationTest() {
+
+    override fun dispatchApiRequest(request: RecordedRequest): MockResponse {
+        return if (request.isRequestBodyEqualTo(createSendMessageRequestJsonString())) {
+            MockResponse()
```

Comment:
The MockResponse().setResponseCode([some int]).setBody("Some string") and this else this `IllegalStateException` will be repeated in every test. You can make the user provide just a predicate and a Pair with response code and responde body then you create the `MockResponse` inside the`BaseMockWebServerIntegrationTest`.